src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -63,15 +63,7 @@ def load_arguments(self, _):
     extension_instance_name_type = CLIArgumentType(help="Name of extension instance, which can be customized. Default: name of the extension.")
     image_template_name_type = CLIArgumentType(overrides=name_arg_type, id_part='name')
     disk_encryption_set_name = CLIArgumentType(overrides=name_arg_type, help='Name of disk encryption set.', id_part='name')
-
-    if self.supported_api_version(min_api='2024-03-01', resource_type=ResourceType.MGMT_COMPUTE):
-        ephemeral_placement_type = CLIArgumentType(options_list=['--ephemeral-os-disk-placement', '--ephemeral-placement'],
-                                                   arg_type=get_enum_type(['ResourceDisk', 'CacheDisk', 'NvmeDisk']),
-                                                   min_api='2024-03-01')
-    else:
-        ephemeral_placement_type = CLIArgumentType(options_list=['--ephemeral-os-disk-placement', '--ephemeral-placement'],
-                                                   arg_type=get_enum_type(['ResourceDisk', 'CacheDisk']),
-                                                   min_api='2019-12-01')
+    ephemeral_placement_type = CLIArgumentType(options_list=['--ephemeral-os-disk-placement', '--ephemeral-placement'], arg_type=get_enum_type(self.get_models('DiffDiskPlacement')), min_api='2019-12-01')
```

Comment:
May I ask why to change this part of code in this PR, since it seems ephemeral_placement_type isn't being used.

Reply:
If just update the arg_type to self.get_models('DiffDiskPlacement'), the newly added option NvmeDisk will be added to the Azure Stack as well. However, it is only accessable after 2024-03-01 API version.

Reply:
The value of DiffDiskPlacement will also change with version
![image](https://github.com/Azure/azure-cli/assets/37168047/fd62f813-0014-4ff4-ba3f-7beef83a9acd)
![image](https://github.com/Azure/azure-cli/assets/37168047/2bf8cea4-9881-49b1-a1fa-9bb76706530c)

Reply:
![image](https://github.com/Azure/azure-cli/assets/37168047/40561ca8-340c-444b-9d53-91fdeae16831)
![image](https://github.com/Azure/azure-cli/assets/37168047/e4c80d17-0a85-4f0b-a7bd-e865c5ac7d1f)